### PR TITLE
Add rest-api collection

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/20991

Docs: https://github.com/giantswarm/giantswarm/pull/21015

For AWS MCs that need SSO the extra rest-api-collection is created via the `gs-rest-api` provider.

For KVM its added my default as we always need the REST API.

```
k get kustomization -n flux-giantswarm
NAME                  READY     STATUS                                                              AGE
collection            True      Applied revision: master/c98735e0450ddffc558ccbd2e80ee49283941acd   11d
customer-flux         True      Applied revision: main/81dd240b7cc4817d92c79fed17b992fad96ea318     11d
flux                  True      Applied revision: main/81dd240b7cc4817d92c79fed17b992fad96ea318     11d
rest-api-collection   Unknown   reconciliation in progress
```

### CODE REVIEW

File: Makefile.custom.mk [hunk 1]
```diff
@@ -14,27 +14,28 @@ manifests/flux-app: FLUXAPP_VERSION := v0.7.1
...
-BOOTSRAP_DEPS += bootstrap/gs-gcp/gs-gcp.yaml
-BOOTSRAP_DEPS += bootstrap/gs-kvm/gs-kvm.yaml
-BOOTSRAP_DEPS += bootstrap/gs-openstack/gs-openstack.yaml
-BOOTSRAP_DEPS += bootstrap/gs-openstack-galaxy/gs-openstack-galaxy.yaml
-BOOTSRAP_DEPS += bootstrap/gs-openstack-gamma/gs-openstack-gamma.yaml
-BOOTSRAP_DEPS += bootstrap/gs-openstack-gravity/gs-openstack-gravity.yaml
-bootstrap: $(BOOTSRAP_DEPS)
+	find manifests/provider/ -wholename '*/kustomization.yaml' | grep -v 'charts' | xargs -I{} gsed -i "0,/version:/{s/version: .*/version: $(FLUXAPP_VERSION)/g}" {}
+
+BOOTSTRAP_DEPS :=
```
Comment (rossf7) [inline Makefile.custom.mk:19]: The typo was killing me 😂

**Comment by rossf7** [comment]
Tested on gorgoth

**Review by anvddriesch — APPROVED** [review]
Very nice! :)

File: Makefile.custom.mk (hunk 1 above)
Comment (piontec) [reply Makefile.custom.mk:19]: I didn't notice :D

File: Makefile.custom.mk [hunk 2]
```diff
@@ -14,27 +14,28 @@ manifests/flux-app: FLUXAPP_VERSION := v0.7.1
...
+BOOTSTRAP_DEPS += bootstrap/customer-aws/customer-aws.yaml
```
Comment (piontec) [inline Makefile.custom.mk:20]: Just for clean ups: `customer-openstack` is (and was already) listed twice

**Review by piontec — APPROVED** [review]
LGTM

File: Makefile.custom.mk (hunk 2 above)
Comment (rossf7) [reply Makefile.custom.mk:20]: Good catch, fixed.

**Pull request merged**